src/WebJobs.Script.WebHost/Scale/TableStorageScaleMetricsRepository.cs

```diff
@@ -293,85 +276,54 @@ internal static TableOperation CreateMetricsInsertOperation(ScaleMetrics metrics
+                {
+                    results.Add(result);
+                }
+
+                return results;
             }
-            catch (StorageException e)
+            catch (RequestFailedException e)
             {
-                if (IsNotFoundTableNotFound(e))
+                if (IsTableNotFound(e))
                 {
-                    return Enumerable.Empty<DynamicTableEntity>();
+                    return Enumerable.Empty<TableEntity>();
                 }
 
                 throw;
             }
         }
 
-        internal async Task<IEnumerable<DynamicTableEntity>> ReadRecentMetrics(CloudTable metricsTable)
+        internal async Task<IEnumerable<TableEntity>> ReadRecentMetrics(TableClient metricsTable)
         {
             // generate a query that will return the most recent metrics
             // based on the configurable max age
             string hostId = await _hostIdProvider.GetHostIdAsync(CancellationToken.None);
             var cuttoff = DateTime.UtcNow - _scaleOptions.ScaleMetricsMaxAge;
             var ticks = string.Format("{0:D19}", DateTime.MaxValue.Ticks - cuttoff.Ticks);
-            string filter = TableQuery.CombineFilters(
-                TableQuery.GenerateFilterCondition(nameof(TableEntity.PartitionKey), QueryComparisons.Equal, hostId),
-                TableOperators.And,
-                TableQuery.GenerateFilterCondition(nameof(TableEntity.RowKey), QueryComparisons.LessThan, ticks));
-            var query = new TableQuery().Where(filter);
-
-            return await ExecuteQuerySafeAsync(metricsTable, query);
-        }
-
-        private async Task<List<DynamicTableEntity>> ExecuteQueryWithContinuationAsync(CloudTable metricsTable, TableQuery query)
-        {
-            List<DynamicTableEntity> results = new List<DynamicTableEntity>();
-            TableContinuationToken continuationToken = null;
-
-            do
-            {
-                var result = await metricsTable.ExecuteQuerySegmentedAsync(query, continuationToken);
-                continuationToken = result.ContinuationToken;
-                results.AddRange(result.Results);
-            }
-            while (continuationToken != null);
-
-            return results;
-        }
-
-        private async Task<IEnumerable<CloudTable>> ListMetricsTablesAsync()
-        {
-            List<CloudTable> tables = new List<CloudTable>();
-            TableContinuationToken continuationToken = null;
-
-            do
-            {
-                var results = await TableClient.ListTablesSegmentedAsync(TableNamePrefix, continuationToken);
-                continuationToken = results.ContinuationToken;
-                tables.AddRange(results.Results);
-            }
-            while (continuationToken != null);
-
-            return tables;
+            return await ExecuteQuerySafeAsync(metricsTable, entity => entity.PartitionKey == hostId && string.Compare(entity.RowKey, ticks) < 0);
```

Comment:
How often is this invoked? If its ran often enough, we should consider formatting a query by string to avoid the expression parsing.

Reply:
It's fairly often. I replaced it with the string flavor using the `CreateQueryFilter` helper.